pkgs/by-name/gh/ghc_filesystem/package.nix

```diff
@@ -19,6 +19,8 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ cmake ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=character-conversion";
```

Comment:
Nit
```suggestion
  # https://github.com/NixOS/nixpkgs/issues/451580
  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang " -Wno-error=character-conversion";
```

Reply:
pushed the version using `lib.optional` to make it a list. let me know if it's better to use the string-based version and I'll change it back to that